hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+
 
 # Theory of Operations
 
-## Block Diagram
+Conceptually speaking, the OTP functionality is at a high level split into "front-end" and "back-end".
+The "front-end" contains the logical partitions and associated buffer registers that feed the hardware and software consumer interfaces of the system.
+The "back-end" represents the programming interface used by hardware and software components to stage the upcoming values.
+The diagram below illustrates this behavioral model.
+
+![OTP Controller Block Diagram](otp_behavioral_model.svg)
+
+Note that the front-end contains both buffered and unbuffered partitions.
+Buffered partitions are sensed once per power cycle and their contents are stored in registers, whereas unbuffered partitions are read on-demand.
+The former are typically partitions that contain data like hardware configuration bits, key material and the life cycle state that need to be always available to the hardware, whereas the latter are large partitions that are accessed infrequently, such as the software configurations.
+Values that are programmed into a buffered partition via the programming interface (coupled with read verification) are merely "staged", and do not take effect until the next power cycle.
+
+The sections below describe the operation of various pieces of the OTP controller and how it supports the described functionality.
+
+## Logical Partitions
+
+The OTP is logically separated into partitions that represent different functions.
+This means the isolation is virtual and maintained by the OTP controller instead of the underlying OTP IP.
+
+Within each logical partition, there are specific enforceable properties
+
+- Secrecy
+  - This controls whether a particular partition contains secret data
+  - If secret, a partition is not readable by software once locked, and is encrypted in storage
+- Read lockability
+  - This controls whether a particular partition disables software readability for later stage software.
+  - Some partitions can be locked statically (by computing and storing an associated digest in OTP), others can be read locked at runtime via a CSRs.
+- Write lockability
+  - This controls whether a partition is locked and prevented from future updates.
+  - A locked partition is stored alongside a 32b digest (truncated hash) to be used later for integrity verification.
+- Integrity Verification
+  - Once a partition is locked, it can undergo periodic verification (time-scale configurable by software)
+This verification takes two forms, partition integrity checks, and storage consistency checks.
+
+Since the OTP is memory-like in nature (it only outputs a certain number of bits per address location), some of the logical partitions are buffered in registers for instantaneous and parallel access by hardware.
+This is a critical point, as after power-up, OTP contents are stored in flip flops and sourced to the system.
+They are **NOT** directly sourced from the OTP macro itself.
```

Comment:
o this line is probably also legacy, should probably clarify software configuration bits are sourced from the OTP directly.

Reply:
ah, yes let me rephrase to make it more clear. this line should only refer to the buffered partitions.